jwst/srctype/srctype_step.py

```diff
@@ -21,25 +22,41 @@ class SourceTypeStep(Step):
 
     spec = """
         source_type = option('POINT','EXTENDED', default=None)  # user-supplied source type
-    """ # noqa: E501
-
-    def process(self, input):
+    """
 
+    def process(self, step_input):
+        """
+        Determine the source type.
+
+        Parameters
+        ----------
+        step_input : str or datamodel
+            Either the path to the file or the science data model to be corrected. The
+            datamodel can be one of the following:
+            stdatamodels.jwst.datamodels.ifuimage.IFUImageModel,
+            stdatamodels.jwst.datamodels.multislit.MultiSlitModel,
+            stdatamodels.jwst.datamodels.slit.SlitModel
```

Comment:
You can put the datamodel options into the type string, e.g.

```suggestion
        step_input : str, IFUImageModel, MultiSlitModel, or SlitModel
            Either the path to the file or the science data model to be corrected.
```